core/i18n.py

```diff
@@ -70,6 +70,9 @@ def __init__(self, key: str, defaultText: str = None, hint: str = None):
             meanings in the target language.
         """
         super(translate, self).__init__()
+
+        if not isinstance(key,str):
```

Comment:
I think the isinstance-test before is slower that just converting all to str. A key, which is already a str, won't be converted as str() itself has an equivalent behavior.
```suggestion
        key = str(key)  # ensure key is a a str
```